src/daemon/rpc_command_executor.cpp

```diff
@@ -2274,31 +2274,58 @@ bool t_rpc_command_executor::sync_info()
     uint64_t target = res.target_height < res.height ? res.height : res.target_height;
     tools::success_msg_writer() << "Height: " << res.height << ", target: " << target << " (" << (100.0 * res.height / target) << "%)";
     uint64_t current_download = 0;
+    uint32_t longest_host_width = 15;
     for (const auto &p: res.peers)
+    {
       current_download += p.info.current_download;
+      if(p.info.address.length() > longest_host_width)
+        {
+          longest_host_width = p.info.address.length();
+        }
+    }
+    longest_host_width += 8;
     tools::success_msg_writer() << "Downloading at " << current_download << " kB/s";
     if (res.next_needed_pruning_seed)
       tools::success_msg_writer() << "Next needed pruning seed: " << res.next_needed_pruning_seed;
 
     tools::success_msg_writer() << std::to_string(res.peers.size()) << " peers";
-    tools::success_msg_writer() << "Remote Host                        Peer_ID   State   Prune_Seed          Height  DL kB/s, Queued Blocks / MB";
+    std::string table_line = "+" + std::string(longest_host_width - 1,'-')+ "+" + std::string(20 - 1,'-') + "+" + std::string(20 - 1,'-') + "+" + std::string(15 - 1,'-') + "+" + std::string(15 - 1,'-') + "+" + std::string(15 - 1,'-') + "+" + std::string(15 - 1,'-') + "+" + std::string(15 - 1,'-') + "+";
```

Comment:
I find the style here quite brittle: To change a column width, you have to edit three separate lines, and they don't even use the same constant (first two are X, last one is X - 1). It'd be way easier to read, and less error-prone, to use a structured array/vector here (e.g. heading + width, where width defaults to size, and can be adjusted for columns with wide content, like `remote host`); such an approach would also make it much easier to wrap the table across lines to fit the terminal, if someone wanted.